config/keys.go

```diff
@@ -189,17 +189,29 @@ func Modules() ([]module.Module, error) {
 		return m, nil
 	}
 
-	if l := len(options); l > 0 {
-		log.
-			WithFields(log.Fields{
-				"options": options,
-			}).
-			Warnf(
-				"Found %d options via flag, but modules are being loaded from configuration file. Ignoring options.",
-				len(options),
-			)
+	modules := file.Modules()
+	if len(modules) == 0 {
+		return modules, errors.New("No modules provided")
 	}
 
-	// TODO: specifying zero modules should be an error (we should add a test for this)
-	return file.Modules(), nil
+	return modulesWithOptions(modules, options), nil
+}
+
+// modulesWithOptions applies command line options to each module and overwrites conflicting options.
+func modulesWithOptions(modules []module.Module, options map[string]interface{}) []module.Module {
+	if len(options) == 0 {
+		return modules
+	}
+
+	for index := range modules {
+		if modules[index].Options == nil {
```

Comment:
This logic is a bit obfuscated. I think the actual _intended_ semantics here are to `make` a map if `nil`.